documentation/AuthServer.md

```diff
@@ -34,7 +35,9 @@ The document types used by the authorization server include:
 | `name` | String | The client name. | None |
 | `oauth2ClientId` | String | The unique identifier issued to a registered OAuth2 client by the authorization server. | None |
 | `oauth2ClientSecret` | String | Confidential shared secret between the client and the authorization server, used to verify the identity of the client. | None |
-| `redirectUri` | String | The URL on client's site where users will be sent after authorization. | None |
+| `oauth2ClientType` | String | The type of clients that use different OAuth2 grant type. GPII supports 4 type of clients: <br>`nativeApp`: native GPII apps that use OAuth2 resource owner password credentials grant.<br>`webApp`: web applications.<br>`thirdPartyApp`: third party applications that GPII users can define privacy settings for. <br>`clientCredentialsApp`: applications that use OAuth2 client credentials grant to add new preferences on behalf of users, such as First Discovery Tool.| Must be one of these values: "nativeApp", "webApp", "thirdPartyApp", "clientCredentialsApp".|

```

Comment:
Simon and I discussed the proposed changes with Antranig at [here](https://botbot.me/freenode/fluid-work/2017-06-23/?msg=87674629&page=1). One extra change from this discussion is to use a specific grant name called "resource owner gpii token grant" instead of OAuth2 conventional "resource owner password credentials grant" for handling the use cases that authorize local flow managers and android apps such as "Mobile Accessibility". Because in these use cases, the user credential that is expected to be received at the Cloud is GPII tokens instead of usernames and passwords.